src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -378,6 +379,7 @@ public void setAutoCommit(boolean autoCommit) throws SQLException
    {
       delegate.setAutoCommit(autoCommit);
       isAutoCommit = autoCommit;
+      isCommitStateDirty = false;
```

Comment:
This is not a valid change.  The JavaDoc says:

> If this method is called during a transaction and the auto-commit mode is changed, the transaction is committed
> If setAutoCommit is called and the auto-commit mode is not changed, the call is a no-op.

If auto-commit started as ``false`` and the user performed some SQL, then ``isCommitStateDirty`` is ``true``.  If the user calls ``setAutoCommit(false)`` during the transaction, it is a no-op (auto-commit is still *false*).  Which means that ``isCommitStateDirty`` should *still* be ``true`` also.  This change would set it to ``false`` unconditionally.